### ISSUE TITLE
The question about shuffle

### ISSUE THREAD

**Issue by staer-tan** [nlpyang/PreSumm#77]
Hi! I find a problem about the data shuffle where the order of loading train dataset always is same sequence. For exmple, it starts cnndm.train.123.bert.pt, cnndm.train.91.bert.pt and cnndm.train.39.bert.pt...every time!

It likely means that it reduce the randomness in the training process. So I wonder how can we do to train the model in fully random?

**Comment by Colanim** [comment]
Data shuffle is random. 

Randomness is guided by the seed.

By default, seed is always `666`. You can modify it by using the option `-seed`, for example `-seed 42`

**Comment by fatmalearning** [comment]
before test i prepared my dataset to get bert file .pt  , i checked it , it the same order of files . when run test, the result generated it log file " gold and candidate " has not the same order 
how can i fix this issue

**Comment by Colanim** [comment]
Gold and candidate files are written together here :

https://github.com/nlpyang/PreSumm/blob/ce8dc017fbef7c12b1b4bd764f0c3d20911ead5e/src/models/predictor.py#L175-L177

So it should not be possible to have wrong order, unless you modified the code.

**Comment by fatmalearning** [comment]
these lines already exist by my version

**Comment by fatmalearning** [comment]
I mean my bert file (.test.0.bert.pt) has the order of the file that will be test, after the testing, the generated log files (gold and candidate ) has not the same order of the bert file (.test.0.bert.pt)
 how can i solve this issue ?
thanks

**Comment by Colanim** [comment]
I think that's why 3 files are written : `gold`, `candidate`, and `source`. These files are the same order.

So if you want to access the article, just refer to the `source` file.

Is it what you need ?

**Comment by fatmalearning** [comment]
yes these 3 source files (gold, candidate, and source) are the same order but all all of them are not the same order of bert file "test.0.pt"

**Comment by zide05** [comment]
same question, when loading test data, the shuffle are set to False, then how could the inference result is shuffled?

**Comment by morningmoni** [comment]
The batch is sorted (not shuffled) by tgt/src length when [creating_batches(self)](https://github.com/nlpyang/PreSumm/blob/master/src/models/data_loader.py#L253) even if the outer loop is not shuffled. Took me a while to figure it out...

**Comment by zhenduow** [comment]
> The batch is sorted (not shuffled) by tgt/src length when [creating_batches(self)](https://github.com/nlpyang/PreSumm/blob/master/src/models/data_loader.py#L253) even if the outer loop is not shuffled. Took me a while to figure it out...

OMG! That saves my day. I got so confused by their output batching order. Now I see it.
So I guess this is the answer to #181
